core/src/main/java/hudson/cli/GroovyshCommand.java

```diff
@@ -75,6 +77,9 @@ protected int run() {
         }
 
         Groovysh shell = createShell(stdin, stdout, stderr);
+
+        LOGGER.log(Level.INFO, "Administrative Groovy script shell spawned");
```

Comment:
You could capture some information which could help to capture what is executed and why. E.g. you can capture user and some other session details, same as in the log below